### ISSUE TITLE
Mobile friendly layout

### ISSUE THREAD

**Issue by rafalborczuch** [git/git-scm.com#1084]
The page rendering is not suited for mobile devices. On a smartphone display the page renders very small font. It is inconvenient to read the doc away from your laptop/PC.

I would expect at least some reasonable font size and the text column stretching to cover the viewport.

**Comment by pedrorijo91** [comment]
Thanks for reporting the problem @rafalborczuch!

Indeed the website may not be optimized for mobile. It may be a big change, but maybe we can introduce small changes incrementally.

Not sure when will this be tackled, but if you are willing to help let us know, and open a PR :)

**Comment by peff** [comment]
Yes, I think it would be nice if the design worked better on a variety of screens. @jasonlong might have some thoughts there, but I think the main issue is just anybody finding time to do a widespread revamp of the site's design.

**Comment by codebangz** [comment]
I would love to help redesign the site to make it responsive on both desktop and mobile.

**Comment by pedrorijo91** [comment]
Nice @codebangz ! :)

For any redesign, we should look at some old issues that should be covered also. For instance: https://github.com/git/git-scm.com/issues/359

**Comment by codebangz** [comment]
Doesn't matter. I can see that orange is the main colour of Git-Scm, we can make the colour more deeper so that users can see the wordings clearly.

**Comment by jasonlong** [comment]
I'm all for folks helping out on the responsive changes. I've begun that project a couple of times since designing the site (almost 6 years ago!) and just never had time to see it through. I'd be happy to oversee these changes though.

Re: the orange – we definitely don't want to change the brand color. The larger orange text labels are probably fine as is, but the uses of orange in code examples should just be replaced with a gray font for better contrast.

**Comment by pedrorijo91** [comment]
so maybe we should start by listing some points we want to tackle? And any other points we find relevant?

**Comment by codebangz** [comment]
Exactly @pedrorijo91

**Comment by pedrorijo91** [comment]
so, any suggestions to start @codebangz @rafalborczuch @jasonlong (and possibly @peff @sxlijin @jnavila) ?

**Comment by jasonlong** [comment]
There's a good bit of general CSS cleanup and refactoring that would be nice to tackle so as to have a cleaner foundation to work on. [This PR](https://github.com/git/git-scm.com/pull/526) is nearly 3 years old and it was never merged because the site was somewhat unmaintained at that point, but I was trying to do some of this cleanup work then.

This is really old (nearly 5 years), but this is the only other thing I ever mocked up for a mobile version of the site. It's not well thought out, but putting here for reference/posterity.

![gitscm-mobile](https://user-images.githubusercontent.com/6104/33836837-d05110b4-de47-11e7-806e-bedd492f21a0.png)

**Comment by pedrorijo91** [comment]
another UX/UI suggestion:

The sidebar sub items should be bold (or any other visual clue), so that a user knows it's in that section.

for instance, under https://git-scm.com/docs, the `Reference` entry should provide some visual clue.

**Comment by sxlijin** [comment]
Suggested wants (and note when I say "analysis" I really just mean "someone
take a look at this"):

- Assess use cases. I don't know what this would look like - perhaps posts
on HN or Reddit asking about what content on git-scm.com people check from
their phones/tablets, maybe just informal coworker polls. Definitely want
more reach than the git mailing list though, where we have people that
actually write the docs.
- Analysis of how tightly the templates and rendering code are coupled.
@jnavila and @peff know more about this than me, but e.g. do we have any
concerns about whether we have to tie into the Asciidoctor pipeline?
- Analysis of the current template structure. Is converting to responsive a
question of simple CSS, or are there more fundamental issues at hand?

Lastly, and this one is important: we need to **decide what browsers we
want to support**. Whether we have CI/CD for them or not is a different
question, but we need a defined scope so that we know what we can and can't
use, e.g. flexboxes. I've seen conversations about backwards compatibility
guarantees in Git go sideways and topsy turvy in the most ridiculous ways
because there was no clear contract/consensus on the extent to which Git
was supposed to be backwards compatible. (Plus, Git users are a whole other
population - beyond the usual Chrome/FF/Safari/Edge, do we care about, say,
Lynx users?)

**Comment by jnavila** [comment]
I'd rather not have to customize the asciidoctor pipeline which isn't stable yet. OTOH, the generated html is already well decorated with classnames, so plugging good css on it should be possible.

**Comment by pedrorijo91** [comment]
@codebangz @jasonlong what else can we do to help you getting started? :)

**Comment by jasonlong** [comment]
I think it would be useful, as a starting point, to either clean up #526 or start a similar branch for fixing a bunch of the styling debt that exists. I mean, we still have some random haml files for crying out loud 🙈 . I can take a stab at that. 

@peff Are you in charge of deploying changes to prod? My main blocker last time around was that nobody was really tasked with that.

**Comment by sxlijin** [comment]
Yes, @peff is the prod maintainer.

Fortunately at this point I believe there's a daily cron job that pushes to prod from HEAD (see comments on #1025), and it's only when the asciidoctor pipeline gets touched when manual intervention is required.

**Comment by peff** [comment]
@jasonlong Yes, I'm happy to deploy any changes. We also auto-deploy to heroku on merges to master these days, so any merged PRs will get deployed. Do note that there's pretty aggressive caching in front of https://git-scm.com, so it may take up to 4 hours for changes to show up there. You can always visit https://git-scm.herokuapp.com to get the uncached view.

I don't think there's really anybody else on the project right now who's very clueful about design or css, so as far as I'm concerned you have carte blanche to make the site look better. :wink:

**Comment by jasonlong** [comment]
@pedrorijo91 Breaking #526 up into more atomic PRs sounds like a good idea. Most of it probably still applies, but there are certain parts I'd like to reconsider.

@peff I was going to create a project board here to list these tasks out, but I don't have permission currently. Could you add me?

**Comment by jasonlong** [comment]
Ok, I created a project board with some initial tasks: https://github.com/git/git-scm.com/projects/1

I took the first couple involving the task runner and linting setup.

**Comment by pedrorijo91** [comment]
awesome @jasonlong !

now it should be way easier to have contributions <3

**Comment by pedrorijo91** [comment]
@codebangz interested in helping out? :) Your help would be much appreciated on #1120 and #1124 (or any other item described in https://github.com/git/git-scm.com/projects/1 you'd like to tackle)

**Comment by jasonlong** [comment]
Just a heads up that I chatted with @peff the other day about some concerns I had with this current approach. My fear is that even after getting this better organized and the basic cleanup work completed, the current CSS structure doesn't lend itself well to making responsive.

My current goal is to spike out an example fully responsive page with an all new CSS framework to test my hypothesis that it'll be easier to start fresh than to rework the current code. I've recruited a fellow GitHubber, @pmarsceill, to help me out.

I don't want to tell anyone not to continue the current cleanup, but be aware that it may not be super useful in the long run.
